### PR TITLE
Tidy up yamls in calicoctl/test-data and improve test_convert.py

### PR DESCRIPTION
**Pull request by coutinhop**
## Description

Remove yamls that were not used by any test and organize them by test
(test_convert.py and multi_context-test.go), plus have inputs and
expected outputs for test_convert.py, which now compares them with
the conversion output.
## Todos

- [x] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: calicoctl/tests/st/calicoctl/test_convert.py [hunk 1]
```diff
@@ -25,53 +26,70 @@
...
         rc = calicoctl("convert -o %s -f %s" % (format, filename))
-        rc.assert_no_error()
 
-        # With the converted data to a temp file
-        with open("/tmp/converted", 'w') as f:
-            f.write(rc.output)
+        # Get expected conversion output filename (file in output/ dir with
+        # yaml or json extension)
+        output_filename = (filename.replace("input", "output").split(".")[0]
+                           + "." + format)
```
Comment (mgleung) [inline calicoctl/tests/st/calicoctl/test_convert.py:63]: is there a reason to split out the file format here? I would have thought that `input` wouldn't be a part of any of the file format so you wouldn't need to split it out.

Comment (coutinhop) [reply calicoctl/tests/st/calicoctl/test_convert.py:63]: I'm doing 2 things: replacing the 'input' dir with 'output' and then replacing the extension with format, e.g.:
'test-data/convert/input/v1/bgppeer-global.yaml' -> 'test-data/convert/output/v1/bgppeer-global.json'

Comment (mgleung) [reply calicoctl/tests/st/calicoctl/test_convert.py:63]: Oh my bad, I misread it and thought that the convert command above already converted the filename. This should be fine.

**Review by mgleung — APPROVED** [review]
Just had one small nit but otherwise looks good to me

**Pull request merged**